Add BatchGetEffectiveIamPolicy sample code

haozhang-google opened the pull request:
## Description

BatchGetEffectiveIamPolicy was not moved from [googleapis/java-asset](https://github.com/googleapis/java-asset) to this repo, move to here manually. Here is the original PR - https://github.com/googleapis/java-asset/pull/1325

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [x] Please **merge** this PR for me once it is approved

averikitsch commented on asset/src/main/java/com/example/asset/BatchGetEffectiveIamPolicyExample.java:
```diff
@@ -0,0 +1,55 @@
+import java.util.Arrays;
+
+/** Batch get effective iam policy example. */
+public class BatchGetEffectiveIamPolicyExample {
```
Please add a main or overloaded method to set variables, see example https://github.com/GoogleCloudPlatform/java-docs-samples/blob/82b17738dc54b4cc7bb76b24c26a8674d0788441/asset/src/main/java/com/example/asset/AnalyzeIamPolicyLongrunningBigqueryExample.java#L33

haozhang-google replied:
Added

averikitsch commented on asset/src/main/java/com/example/asset/BatchGetEffectiveIamPolicyExample.java:
```diff
@@ -0,0 +1,55 @@
+      System.out.println("BatchGetEffectiveIamPolicies completed successfully:\n" + response);
+    } catch (IOException e) {
+      System.out.println("Failed to create client:\n" + e);
+    } catch (ApiException e) {
```
Only catch exceptions if you are going to show how to handle them (https://github.com/GoogleCloudPlatform/java-docs-samples/blob/82b17738dc54b4cc7bb76b24c26a8674d0788441/SAMPLE_FORMAT.md#exception-handling)

haozhang-google replied:
Removed catch exceptions.

haozhang-google replied:
Hi @averikitsch , the solution to solve IOException or ApiException is too verbose to solve in a sample, therefore I just leave the log per - https://github.com/GoogleCloudPlatform/java-docs-samples/blob/82b17738dc54b4cc7bb76b24c26a8674d0788441/SAMPLE_FORMAT.md#exception-handling. Also you can see our other examples here - https://github.com/GoogleCloudPlatform/java-docs-samples/blob/82b17738dc54b4cc7bb76b24c26a8674d0788441/asset/src/main/java/com/example/asset/AnalyzeIamPolicyLongrunningBigqueryExample.java#L71-L82, we also just leave a log for such exceptions. Thanks

averikitsch commented:
The method does need to throw the error:
```
[ERROR] COMPILATION ERROR :
[ERROR] /tmpfs/src/github/java-docs-samples/asset/src/main/java/com/example/asset/BatchGetEffectiveIamPolicyExample.java:[38,33] unreported exception java.lang.Exception; must be caught or declared to be thrown
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project cloudasset-snippets: Compilation failure
[ERROR] /tmpfs/src/github/java-docs-samples/asset/src/main/java/com/example/asset/BatchGetEffectiveIamPolicyExample.java:[38,33] unreported exception java.lang.Exception; must be caught or declared to be thrown
```

haozhang-google commented:
> The method does need to throw the error:
> 
> ```
> [ERROR] COMPILATION ERROR :
> [ERROR] /tmpfs/src/github/java-docs-samples/asset/src/main/java/com/example/asset/BatchGetEffectiveIamPolicyExample.java:[38,33] unreported exception java.lang.Exception; must be caught or declared to be thrown
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project cloudasset-snippets: Compilation failure
> [ERROR] /tmpfs/src/github/java-docs-samples/asset/src/main/java/com/example/asset/BatchGetEffectiveIamPolicyExample.java:[38,33] unreported exception java.lang.Exception; must be caught or declared to be thrown
> ```

This is because I removed the exception handling part before, now added back. Justifications are written in above comment, thanks.
